### PR TITLE
overlord: start with a skeleton and stubs for Overlord, StateEngine, StateJournal and managers

### PR DESCRIPTION
**Pull request by pedronis**
This starts with a skeleton and stubs for Overlord, StateEngine, StateJournal and snap, assertion and skill managers and State and other state management related types.

### CODE REVIEW

File: overlord/overlord.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+type Overlord struct {
+	// managers
+	snapMgr   *SnapManager
+	assertMgr *AssertManager
+	skillMgr  *SkillManager
+}
+
+// New creates a new Overlord with all its state managers.
+func New() (*Overlord, error) {
+	stateEng := NewStateEngine()
```
Comment (zyga) [inline overlord/overlord.go]: Don't you want to store that somewhere?

**Comment by zyga** [comment]
+1 with the exception of the question above.

**Comment by zyga** [comment]
Question: where are unit test results? Is travis having a hiccup?

File: overlord/overlord.go (hunk 1 above)
Comment (pedronis) [inline overlord/overlord.go]: yes, we need to pass it when we make the StateJournal, something I have postponed

**Comment by mvo5** [comment]
Looks great, thanks for doing this!

**Pull request merged**